### PR TITLE
Removing clipped{sigmoid|softmax} since baseline pytorch primitives are stable enough

### PR DESCRIPTION
**Pull request by rohitsingh0812**
I tested SS-VAE for default configuration and it converged to a good accuracy without the Clipped operations

<details>

(pyro-dev-py3.6) rohits@rohits-desktop:~/Uber/pyro-main/examples/vae$ python ss_vae_M2.py --cuda -alm 46 -sup 3000 -zd 50 -hl 500 -lr 0.0042 -log ./M2_3000_b.log -enum parallel
0 epoch: avg losses 290.9319388020833 80.50945426432291 189.17907845744682 0.0 validation accuracy 0.8666 test accuracy 0.8539
1 epoch: avg losses 150.44911458333334 20.916288736979165 146.3790541057181 0.0 validation accuracy 0.9164 test accuracy 0.9074
2 epoch: avg losses 133.376736328125 11.97279581705729 130.77526238364362 0.0 validation accuracy 0.9333 test accuracy 0.9264
3 epoch: avg losses 124.29458984375 7.431863749186198 122.26971766954787 0.0 validation accuracy 0.9448 test accuracy 0.9382
4 epoch: avg losses 119.38181901041666 3.6734229787190755 117.64761664727394 0.0 validation accuracy 0.9434 test accuracy 0.936
5 epoch: avg losses 116.50273893229166 2.9408125330607096 114.88073749168883 0.0 validation accuracy 0.9463 test accuracy 0.9402
6 epoch: avg losses 114.21403190104166 2.030322146097819 112.97056491023936 0.0 validation accuracy 0.9513 test accuracy 0.9484
7 epoch: avg losses 112.28866796875 1.0412305119832357 111.09174867021277 0.0 validation accuracy 0.9541 test accuracy 0.9487
8 epoch: avg losses 111.10883203125 1.0258033487002054 109.72927198304521 0.0 validation accuracy 0.9547 test accuracy 0.9521
9 epoch: avg losses 109.74407747395833 0.5513597167332968 108.60308132480053 0.0 validation accuracy 0.9538 test accuracy 0.951
10 epoch: avg losses 109.06003450520834 1.9920017115275066 107.81676030585106 0.0 validation accuracy 0.9598 test accuracy 0.9555
11 epoch: avg losses 107.96032096354166 0.9248500928878784 107.05213123337766 0.0 validation accuracy 0.9576 test accuracy 0.9546
12 epoch: avg losses 107.51856510416667 2.2424600923061373 106.45374393284574 0.0 validation accuracy 0.9577 test accuracy 0.9527
13 epoch: avg losses 106.78620052083333 1.671778155485789 105.83291356382979 0.0 validation accuracy 0.9604 test accuracy 0.9602
14 epoch: avg losses 106.35078971354167 1.0128531044721603 105.29195665724734 0.0 validation accuracy 0.962 test accuracy 0.9609
15 epoch: avg losses 105.79403190104166 1.643526054215928 104.89332255651595 0.0 validation accuracy 0.9583 test accuracy 0.9549
16 epoch: avg losses 105.36958463541667 2.518962747414907 104.73857662898936 0.0 validation accuracy 0.9563 test accuracy 0.9555
17 epoch: avg losses 104.96056119791666 1.343245118935903 104.24255049035904 0.0 validation accuracy 0.9623 test accuracy 0.9604
18 epoch: avg losses 104.809669921875 0.8844728315671285 104.02976034740692 0.0 validation accuracy 0.9634 test accuracy 0.9644
19 epoch: avg losses 104.594056640625 0.6874861124927799 103.76789349235372 0.0 validation accuracy 0.9639 test accuracy 0.9609
20 epoch: avg losses 104.30488932291667 0.5577879299074412 103.55861751994681 0.0 validation accuracy 0.9588 test accuracy 0.9613
21 epoch: avg losses 104.26286393229167 1.7229827137838196 103.5112753075133 0.0 validation accuracy 0.9586 test accuracy 0.9585
22 epoch: avg losses 103.95945963541666 1.6902066664497057 103.24549193816489 0.0 validation accuracy 0.9596 test accuracy 0.9606
23 epoch: avg losses 103.92058658854167 2.094724974832808 103.1933181931516 0.0 validation accuracy 0.9618 test accuracy 0.9614
24 epoch: avg losses 103.85938736979166 4.467386712272962 103.17234266954787 0.0 validation accuracy 0.9599 test accuracy 0.9619
25 epoch: avg losses 103.63644401041667 2.2642598028481395 102.87117848238032 0.0 validation accuracy 0.9642 test accuracy 0.9645
26 epoch: avg losses 103.61386783854167 1.7810563671938144 102.70043687666224 0.0 validation accuracy 0.9652 test accuracy 0.9657
27 epoch: avg losses 103.36332161458333 2.581535070801775 102.7066953125 0.0 validation accuracy 0.9594 test accuracy 0.9606
28 epoch: avg losses 103.08237109375 1.766889231924705 102.49495466256649 0.0 validation accuracy 0.9632 test accuracy 0.9661
29 epoch: avg losses 103.10711067708333 2.4985556071147634 102.47751292386968 0.0 validation accuracy 0.9622 test accuracy 0.9649
30 epoch: avg losses 103.047193359375 0.6225819512369344 102.30740126329788 0.0 validation accuracy 0.9594 test accuracy 0.9582
31 epoch: avg losses 102.929353515625 0.7885431037551025 102.20885384807181 0.0 validation accuracy 0.9623 test accuracy 0.9643
32 epoch: avg losses 103.09781966145833 1.5114867982995541 102.22672153424202 0.0 validation accuracy 0.9649 test accuracy 0.9661
33 epoch: avg losses 102.88453059895834 1.1373855056689741 102.12132937167553 0.0 validation accuracy 0.9618 test accuracy 0.9642
34 epoch: avg losses 102.978958984375 1.8480762501970553 102.03723142453457 0.0 validation accuracy 0.9658 test accuracy 0.964
35 epoch: avg losses 102.817958984375 3.128196001511106 101.96273799035905 0.0 validation accuracy 0.9632 test accuracy 0.9638
36 epoch: avg losses 102.4742578125 1.749193319843267 101.86617557347074 0.0 validation accuracy 0.963 test accuracy 0.964
37 epoch: avg losses 102.436837890625 0.8139813101323089 101.71805456283245 0.0 validation accuracy 0.9654 test accuracy 0.966
38 epoch: avg losses 102.43426953125 0.9948910545959371 101.75966019780586 0.0 validation accuracy 0.9664 test accuracy 0.9656
39 epoch: avg losses 102.422359375 1.7882741417044696 101.6328243018617 0.0 validation accuracy 0.9596 test accuracy 0.9615
40 epoch: avg losses 102.344296875 5.0308908309664115 101.696705078125 0.0 validation accuracy 0.9568 test accuracy 0.9592
41 epoch: avg losses 102.42167838541667 2.699498481253822 101.7342222406915 0.0 validation accuracy 0.9612 test accuracy 0.9645
42 epoch: avg losses 102.29066341145834 2.322416580331822 101.49999688331117 0.0 validation accuracy 0.9642 test accuracy 0.9635
43 epoch: avg losses 102.21257877604167 1.8881427197644953 101.54680680684841 0.0 validation accuracy 0.9655 test accuracy 0.965
44 epoch: avg losses 103.14259765625 1.3036428237709916 103.60167528257979 0.0 validation accuracy 0.9676 test accuracy 0.9676

</details>

### CODE REVIEW

File: examples/vae/ss_vae_M2.py [hunk 1]
```diff
@@ -405,13 +396,14 @@ def main(args):
 
     parser.add_argument('--cuda', action='store_true',
                         help="use GPU(s) to speed up training")
-    parser.add_argument('-ne', '--num-epochs', default=100, type=int,
+    parser.add_argument('-ne', '--num-epochs', default=50, type=int,
```
Comment (martinjankowiak) [inline examples/vae/ss_vae_M2.py]: i thought this was already changed to '-n' (consistent with other examples/tutorials)

File: examples/vae/ss_vae_M2.py [hunk 2]
```diff
@@ -405,13 +396,14 @@ def main(args):
...
                         help="number of epochs to run")
     parser.add_argument('--aux-loss', action="store_true",
-                        help="whether to use the auxiliary loss from NIPS 14 paper (Kingma et al)")
-    parser.add_argument('-alm', '--aux-loss-multiplier', default=300, type=float,
+                        help="whether to use the auxiliary loss from NIPS 14 paper "
+                             "(Kingma et al) -- it is not used by default ")
+    parser.add_argument('-alm', '--aux-loss-multiplier', default=46, type=float,
                         help="the multiplier to use with the auxiliary loss")
-    parser.add_argument('-enum', '--enum-discrete', default=None,
+    parser.add_argument('-enum', '--enum-discrete', default="parallel",
```
Comment (martinjankowiak) [inline examples/vae/ss_vae_M2.py]: help message doesn't message default choice

File: examples/vae/ss_vae_M2.py (hunk 1 above)
Comment (jpchen) [inline examples/vae/ss_vae_M2.py]: can you change this back to what it was in dev (i think `-n`) to match the other examples

File: examples/vae/utils/custom_mlp.py [hunk 3]
```diff
@@ -56,24 +55,15 @@ def forward(self, *args, **kwargs):
...
-            return op(epsilon, dim=1)
-        except TypeError:
-            # Support older pytorch 0.2 release.
-            return op(epsilon)
-    elif op in [ClippedSigmoid]:
-        return op(epsilon)
-    elif op in [nn.Softmax, nn.LogSoftmax]:
+    if op in [nn.Softmax, nn.LogSoftmax]:
         return op(dim=1)
     else:
```
Comment (jpchen) [inline examples/vae/utils/custom_mlp.py]: nit: redundant `else`

Comment (fritzo) [reply examples/vae/utils/custom_mlp.py]: Let's refrain from pushing a fix tho. we can fix that after 0.2.0
